azurerm/internal/services/cosmos/cosmosdb_cassandra_table_resource.go

```diff
@@ -0,0 +1,421 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/cosmos/common"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/cosmos/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceCosmosDbCassandraTable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCosmosDbCassandraTableCreate,
+		Read:   resourceCosmosDbCassandraTableRead,
+		Update: resourceCosmosDbCassandraTableUpdate,
+		Delete: resourceCosmosDbCassandraTableDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.CosmosEntityName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.CosmosAccountName,
+			},
+
+			"default_ttl": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntAtLeast(-1),
+			},
+
+			"keyspace_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.CosmosEntityName,
+			},
+
+			"schema": common.CassandraTableSchemaPropertySchema(),
+
+			"throughput": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validate.CosmosThroughput,
+			},
+
+			"autoscale_settings": common.DatabaseAutoscaleSettingsSchema(),
+		},
+	}
+}
+
+func resourceCosmosDbCassandraTableCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Cosmos.CassandraClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	account := d.Get("account_name").(string)
+	keyspace := d.Get("keyspace_name").(string)
+
+	existing, err := client.GetCassandraTable(ctx, resourceGroup, account, keyspace, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("Error checking for presence of creating Cosmos Cassandra Table %q (Account: %q, Keyspace: %q): %+v", name, account, keyspace, err)
```

Comment:
fwiw this can become:

```suggestion
			return fmt.Errorf("checking for presence of existing %+v: %+v", id, err)
```